shared/reducers/fs.js

```diff
@@ -102,6 +110,47 @@ const updatePathItem = (oldPathItem?: ?Types.PathItem, newPathItem: Types.PathIt
   }
 }
 
+const updateTlf = (oldTlf?: ?Types.Tlf, newTlf: Types.Tlf): Types.Tlf => {
+  if (!oldTlf) {
+    return newTlf
+  }
+  // TODO: Ideally this should come in with other data from the same RPC.
+  const newTlfDontClearSyncConfig = newTlf.syncConfig ? newTlf : newTlf.set('syncConfig', oldTlf.syncConfig)
+  if (!newTlf.resetParticipants.equals(oldTlf.resetParticipants)) {
+    return newTlfDontClearSyncConfig
+  }
+  if (!I.is(newTlfDontClearSyncConfig.waitingForParticipantUnlock, oldTlf.waitingForParticipantUnlock)) {
+    return newTlfDontClearSyncConfig
+  }
+  if (!I.is(newTlfDontClearSyncConfig.youCanUnlock, oldTlf.youCanUnlock)) {
+    return newTlfDontClearSyncConfig
+  }
+  if (
+    !I.is(newTlfDontClearSyncConfig.syncConfig, oldTlf.syncConfig) &&
+    !(
+      newTlfDontClearSyncConfig.syncConfig &&
+      oldTlf.syncConfig &&
+      newTlfDontClearSyncConfig.syncConfig.mode === 'partial' &&
+      oldTlf.syncConfig.mode === 'partial' &&
+      newTlfDontClearSyncConfig.syncConfig.enabledPaths.equals(oldTlf.syncConfig.enabledPaths)
+    )
+  ) {
+    return newTlfDontClearSyncConfig
+  }
+  return oldTlf.merge(
+    newTlfDontClearSyncConfig.withMutations(n =>
+      n
+        .set('resetParticipants', oldTlf.resetParticipants)
+        .set('waitingForParticipantUnlock', oldTlf.waitingForParticipantUnlock)
+        .set('youCanUnlock', oldTlf.youCanUnlock)
+        .set('syncConfig', oldTlf.syncConfig)
+    )
+  )
+}
+
+const updateTlfList = (oldTlfList: Types.TlfList, newTlfList: Types.TlfList): Types.TlfList =>
+  newTlfList.withMutations(list => list.map((tlf, name) => updateTlf(oldTlfList.get(name), tlf)))
```

Comment:
I don't think `withMutations` helps us here, since there's only one operation underneath it.